core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -188,11 +189,28 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
             asInstanceOf[java.util.ArrayList[SnappyRegionStatsCollectorResult]]
             .asScala
       }
+
+      // External Tables
+      val hiveTables: java.util.List[ExternalTableMetaData] =
```

Comment:
Can be simplified. 
`val externalTables = Misc.getMemStore.getExternalCatalog.getHiveTables(true).asScala.filter(extTable => extTable.tableType.equalsIgnoreCase("EXTERNAL")).map(extTable => new SnappyExternalTableStats (extTable.entityName, extTable.tableType, extTable.provider, extTable.externalStore, extTable.dataSourcePath, extTable.driverClass))`

Reply:
Or one step more with collect :)

```scala
hiveTables.asScala.collect {
  case table if table.tableType.equalsIgnoreCase("EXTERNAL") =>
      new SnappyExternalTableStats(...)
}
```

Reply:
Updated code as per Sumedh's suggestion.